src/mudlet.h

```diff
@@ -649,4 +663,43 @@ class TConsoleMonitor : public QObject
 };
 
 
+// A convenience class to keep all the details for the translators for a
+// specific locale code (langauge only "xx" or language/country "xx_YY")
+// in one unified structure.
+class translation
+{
+    // The following must have friendship so they can set private members:
+    friend void mudlet::scanForMudletTranslations(const QString&);
+    friend void mudlet::scanForQtTranslations(const QString&);
+
+public:
+    translation(const int translationPercent = -1) : mTranslatedPercentage(translationPercent) {}
+
+    const QString& getNativeName() const { return mNativeName; }
+    const QString& getMudletTranslationPathFile() const { return mMudletTranslationPathFileName; }
+    const QString& getQtTranslationPathFile() const { return mQtTranslationPathFileName; }
+    const int& getTranslatedPercentage() const { return mTranslatedPercentage; }
+    bool getIsFromResourceFile() const { return mTranslatedPercentage >= 0; }
+
+private:
+    // Used for display in the profile preferences and is never translated:
+    QString mNativeName;
+    // ONLY if the translation is loaded from an embedded resource file,
+    // is the percentage complete of the translation - determined via a lua
+    // script that parses the output of the lrelease executable that
+    // converts the source mudlet_xx_YY.ts files into the binary
+    // mudlet_xx_YY.qm files placed into the embedded resource file during
+    // building the application:
+    int mTranslatedPercentage;
```

Comment:
Not sure why we need to document how it's obtained here. What value does it add to the coder reading it?

Reply:
Because that percentage won't be available (will be `-1`) in all potential future scenarios - for instance if the translation is sourced from elsewhere and not embedded in the resource file. (Packagers will likely do that to keep the translations in say, `\usr\share\mudlet\translations` - indeed in that situation unneeded language/locales can be removed by the end *super* user - see `localepurge(8)`).